File: MovieTicket.cs
class MovieTicket
{
    MovieScreening movieScreening;
    int rowNr;
    int seatNr;
    Boolean isPremium { get; }

    public MovieTicket(MovieScreening movieScreening, int rowNr, int seatNr, Boolean isPremium)
    {
        this.movieScreening = movieScreening;
        this.rowNr = rowNr;
        this.seatNr = seatNr;
        this.isPremium = isPremium;
    }

    public double GetPrice(Boolean isStudentOrder)
    {
        double price = movieScreening.PricePerSeat;
        if (isPremium)
        {
            price += isStudentOrder ? 2 : 3;
        }
    }

    public override string ToString()
    {
        string premiumText = isPremium ? " (Premium Seat)" : "";
        return $"Ticket for {movieScreening}, Row: {rowNr}, Seat: {seatNr}{premiumText}";
    }
}

File: src/main/java/soagroup15a/MovieTicket.java
package src.main.java.soagroup15a;

public class MovieTicket {
    private int rowNr;
    private int seatNr;
    private boolean isPremium;
    private MovieScreening movieScreening;

    public MovieTicket(
            MovieScreening movieScreening,
            boolean isPremiumReservation,
            int seatRow,
            int seatnr) {
        this.movieScreening = movieScreening;
        this.isPremium = isPremiumReservation;
        this.rowNr = seatRow;
        this.seatNr = seatnr;
    }

    public boolean isPremiumTicket() {
        return isPremium;
    }

    public MovieScreening getMovieScreening() {
        return movieScreening;
    }

    @Override
    public String toString() {
        String premiumText = isPremium ? " (Premium)" : "";
        return "Ticket for " + movieScreening + ", Row: " + rowNr + ", Seat: " + seatNr + premiumText;
    }
}